File: src/main/java/com/example/studentmanagementsystem/mappers/MarkMapper.java
package com.example.studentmanagementsystem.mappers;

import com.example.studentmanagementsystem.entity.Mark;
import com.example.studentmanagementsystem.entity.vo.MarkVO;
import java.util.List;
import org.apache.ibatis.annotations.Mapper;
import org.apache.ibatis.annotations.Param;
import org.springframework.stereotype.Repository;

/** @author Hingbong */
@Mapper
@Repository
public interface MarkMapper {

  /**
   * add new mark
   *
   * @param mark mark
   * @return rows affected
   */
  Integer addNewMark(Mark mark);

  /**
   * get sum of final score by id
   *
   * @param stuId student id
   * @return sum of final score
   */
  Double getTotalFinalScoreById(Integer stuId);

  /**
   * find all marks
   *
   * @param addDate the date of adding to the table
   * @param stuName student name
   * @param afterDay the day after add date
   * @return the result
   */
  List<MarkVO> findAll(
      @Param("addDate") String addDate,
      @Param("afterDay") String afterDay,
      @Param("stuName") String stuName);

  /**
   * delete a mark by id
   *
   * @param markId mark id
   * @return rows affected
   */
  Integer deleteMarkById(Integer markId);
}

File: src/main/kotlin/com/example/studentmanagementsystem/mappers/MarkMapper.kt
package com.example.studentmanagementsystem.mappers

import com.example.studentmanagementsystem.entity.Mark
import com.example.studentmanagementsystem.entity.vo.MarkVO
import org.apache.ibatis.annotations.Param
import org.springframework.stereotype.Repository

/** @author Hingbong
 */
@Repository
interface MarkMapper {

    /**
     * add new mark
     *
     * @param mark mark
     * @return rows affected
     */
    fun addNewMark(mark: Mark): Int

    /**
     * get sum of final score by id
     *
     * @param stuId student id
     * @return sum of final score
     */
    fun getTotalFinalScoreById(stuId: Int): Double?

    /**
     * find all marks
     *
     * @param addDate the date of adding to the table
     * @param stuName student name
     * @return the result
     */
    fun findAll(@Param("addDate") addDate: String?, @Param("stuName") stuName: String?): List<MarkVO>

    fun findByStuId(stuId: Int): List<Mark>
    /**
     * delete a mark by id
     *
     * @param markId mark id
     * @return rows affected
     */
    fun deleteMarkById(markId: Int): Int
}
